azurelinuxagent/common/event.py

```diff
@@ -581,24 +580,23 @@ def _trim_extension_event_parameters(event):
         event.parameters = trimmed_params
 
     @staticmethod
-    def report_dropped_events_error(count, errors, op, max_errors_to_report): # pylint: disable=C0103
+    def report_dropped_events_error(count, errors, operation_name, max_errors_to_report):
         err_msg_format = "DroppedEventsCount: {0}\nReasons (first {1} errors): {2}"
         if count > 0:
-            add_event(op=op,
+            add_event(op=operation_name,
                       message=err_msg_format.format(count, max_errors_to_report, ', '.join(errors)),
                       is_success=False)
 
-    def collect_events(self): # pylint: disable=R0914
+    def collect_events(self, enqueue_event_func): # pylint: disable=R0914
         """
         Retuns a list of events that need to be sent to the telemetry pipeline and deletes the corresponding files
         from the events directory.
         """
         max_collect_errors_to_report = 5
-        event_list = TelemetryEventList()
         event_directory_full_path = os.path.join(conf.get_lib_dir(), EVENTS_DIRECTORY)
         event_files = os.listdir(event_directory_full_path)
-        unicode_error_count, unicode_errors = 0, []
-        collect_event_error_count, collect_event_errors = 0, []
+        unicode_error_count, unicode_errors = 0, set()
```

Comment:
At this point we have separate logic to collect events for extensions and the agent. Are you planning on unifying those? It seems like there is logic (e.g. error handling) that should be shared between them.

Reply:
both have somewhat different logics. The extension telemetry code maintains a dict of static error messages and their counts to inform Extension publishers for their breach of contract. This logic though only captures the first 5 errors thrown and are used for debugging from the agent. We can talk more about this offline

Reply:
Kept both logics different as of now, but unified the error reporting for Reporting events and Collecting agent events